darts/models/ensemble_model.py

```diff
@@ -0,0 +1,66 @@
+"""
+Ensemble model
+--------------
+"""
+
+from abc import abstractmethod
+from typing import List, Optional
+
+from ..timeseries import TimeSeries
+from ..logging import get_logger, raise_if_not
+from ..models.forecasting_model import ForecastingModel
+
+logger = get_logger(__name__)
+
+
+class EnsembleModel(ForecastingModel):
+    """
+    Abstract base class for ensemble models.
+    Ensemble models take in a list of forecasting models and ensemble their predictions
+    to make a single one according to the rule defined by their `ensemble()` method.
+    """
+    def __init__(self, models: List[ForecastingModel]):
+        raise_if_not(isinstance(models, list) and models,
+                     "Cannot instantiate EnsembleModel with an empty list of models",
+                     logger)
+        raise_if_not(all(isinstance(model, ForecastingModel) for model in models),
+                     "All models must be instances of darts.models.ForecastingModel",
+                     logger)
+        super().__init__()
+        self.models = models
+
+    def fit(self, training_series: TimeSeries, target_series: Optional[TimeSeries] = None) -> None:
+        super().fit(training_series, target_series)
+
+        for model in self.models:
```

Comment:
Not sure the models should be fit here.
If you take the example of GROE, the GROE function fits them all anyway, so no need to call model.fit() here. In general I think it looks like it should depend on the implementation, and hence not be in the base class.

Reply:
Good catch, I totally missed that this was fitting the models twice.

I think we can indeed remove the `model.fit()` calls from the `EnsembleModel` abstract base class, but I might want to add a note in the `EnsembleModel.fit()` docstring to remember users inherinting from the class that this doesn't call `fit()` on the individual models and that it's the child's responsibility to call do so in its own `fit()` method. What do you think ?

Reply:
Yes, that sounds good!